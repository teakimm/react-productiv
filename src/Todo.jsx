import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({ todo }) {
  return (
    <div>
      {todo.completed === false ?
        <div className="Todo">
          <div><b>{todo.title}</b> <small>({`priority: ${todo.priority}`})</small></div>
          <div><small>{todo.description}</small></div>
        </div>
        :
        <div className="Todo">
          <div><b><s>{todo.title}</s></b> <small>({`priority: ${todo.priority}`})</small></div>
          <div><small>{todo.description}</small></div>
        </div>


      }
    </div>

  );
}

export default Todo;
