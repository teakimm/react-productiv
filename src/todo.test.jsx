import { describe, it, expect, test } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const testTodo = {
    id: "anyId",
    title: "testTitle",
    description: "hello world",
    priority: 3
};

it("renders without crashing", function () {
    render(<Todo todo={testTodo} />);
});

it("matches snapshot", function () {
    const { container } = render(<Todo todo={testTodo} />);
    expect(container).toMatchSnapshot();
});

it("contains information about todo", function () {
    const result = render(<Todo todo={testTodo} />);

    expect(
        result.queryByText("hello world")
    ).toBeInTheDocument();

    expect(
        result.queryByText("testTitle")
    ).toBeInTheDocument();

});

it("does not contain id", function () {
    const result = render(<Todo todo={testTodo} />);

    expect(
        result.queryByText("anyId")
    ).not.toBeInTheDocument();
});





