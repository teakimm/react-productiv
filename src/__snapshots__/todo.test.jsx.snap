// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        testTitle
      </b>
       
      <small>
        (
        priority: 3
        )
      </small>
    </div>
    <div>
      <small>
        hello world
      </small>
    </div>
  </div>
</div>
`;
